### PR TITLE
openshift-upgrade-gcp-loki: persist monitoring metrics via PVC

### PR DESCRIPTION
**Pull request by vrutkovs**
Add a step to setup PVC for monitoring, so that metrics would persist over upgrade - by default Prometheus is using emptyDir, so initial cluster metrics are wiped during upgrade.

With this change CVO upgrade test kept all metrics:
```
2020/09/02 18:15:20 Executing "e2e-upgrade-ipi-install-install"
...
2020/09/02 18:47:58 Pod e2e-upgrade-ipi-install-install succeeded after 32m37s 
...
2020/09/02 18:50:55 Executing "e2e-upgrade-openshift-e2e-test"
...
2020/09/02 20:36:12 Pod e2e-upgrade-ipi-deprovision-deprovision succeeded after 4m46s
```
and PromeCIeus shows metrics for 18:38 - 20:34:

Screenshot-20200903094444-1681x821

### CODE REVIEW

File: ci-operator/step-registry/ipi/gcp/pre/monitoringpvc/ipi-gcp-pre-monitoringpvc-chain.yaml
Comment (wking) [inline ci-operator/step-registry/ipi/gcp/pre/monitoringpvc/ipi-gcp-pre-monitoringpvc-chain.yaml:6]: I'm not a big believer in the pre/post model, when these are so easy to inline in the workloads.  Or push this down?  Which consumers of ipi-install would not want this?

**Comment by wking** [comment]
One nit, and you probably want at least some of the owners you're volunteering to sign off on this.  But I don't have a problem with this landing as it stands.

File: ci-operator/step-registry/ipi/gcp/pre/monitoringpvc/ipi-gcp-pre-monitoringpvc-chain.yaml
Comment (vrutkovs) [reply ci-operator/step-registry/ipi/gcp/pre/monitoringpvc/ipi-gcp-pre-monitoringpvc-chain.yaml:6]: This alters default openshift install - and some platforms may not be able to provide a PV

**Comment by vrutkovs** [comment]
Adding folks from testplatform and monitoring for review

/cc @stevekuznetsov @petr-muller @paulfantom @simonpasquier

File: ci-operator/step-registry/ipi/gcp/pre/monitoringpvc/ipi-gcp-pre-monitoringpvc-chain.yaml
Comment (wking) [reply ci-operator/step-registry/ipi/gcp/pre/monitoringpvc/ipi-gcp-pre-monitoringpvc-chain.yaml:6]: So in that case, I'd push this up to the workflow without a pre chain.  But I don't have a problem with the pre chain if other folks want to maintain it.

**Comment by paulfantom** [comment]
LGTM from monitoring side

**Review by petr-muller — APPROVED** [review]
/approve
/hold

Holding to allow watching after merge, if needed. Feel free to unhold.

**Comment by vrutkovs** [comment]
/hold cancel

Tests are passing, metrics are persisted across CMO deployments

**Comment by jstuever** [comment]
Yikes...this should have landed as `ipi-conf-monitoringvpc`

**Pull request merged**
